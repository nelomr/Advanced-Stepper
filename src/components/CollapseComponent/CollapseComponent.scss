.collapse-component {
    background-color: transparent;

    &.is-full-width {
        padding: 0 20px;
        margin: 0 -20px;

        @media (--min-tablet) {
            padding: 0;
            margin: 0;
        }
    }

    > div {
        display: flex;
        align-items: flex-start;
        justify-content: space-between;
    }

    i {
        display: flex;
        justify-content: center;
        align-items: center;
        font-size: 30px;
        transform: scale(1);
        transform-origin: center;
        transition: transform .3s ease;
    }

    &.is-open {
        i {
            transform: scale(-1);
        }
    }
}

.collapse--button {
    cursor: hand;
    cursor: pointer;
}

.collapse-component--wrapper {
    overflow: hidden;

    .is-full-width & {
        padding: 0 20px;
        margin: 0 -20px;

        @media (--min-tablet) {
            padding: 0;
            margin: 0;
        }
    }

    .is-open & {
        > .collapse-component--collapsible {
            max-height: 100000px;
            margin-bottom: 0;
            visibility: visible;
            transition: margin-bottom .3s cubic-bezier(0, 0, 0, 1);
        }
    }
}

.collapse-component--collapsible {
    width: 100%;
    max-height: 0;
    margin-bottom: -2000px;
    transition: margin-bottom .3s cubic-bezier(1, 0, 1, 1), visibility 0s .3s, max-height 0s .3s;
}

.collapse-component--content {
    @media (--min-tablet) {
        display: flex;
        flex-wrap: wrap;
        justify-content: center;
    }
}
