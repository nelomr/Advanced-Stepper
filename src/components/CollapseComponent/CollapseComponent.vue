<template>
    <div :class="['collapse-component', classes]">
        <div class="collapse-trigger" @click="toggle()">
            <slot name="collapse-component-header" />
        </div>
        <div class="collapse-component--wrapper">
            <div class="collapse-component--collapsible">
                <div class="collapse-component--content">
                    <slot name="collapse-component-content" />
                </div>
            </div>
        </div>
    </div>
</template>

<script>
export default {
    props: {
        openDefault: {
            type: Boolean,
            default: false
        }
    },
    data() {
        return {
            isActive: this.openDefault
        };
    },
    computed: {
        classes() {
            return {
                'is-open': this.isActive
            };
        }
    },
    methods: {
        open() {
            this.isActive = true;
        },
        close() {
            this.isActive = false;
        },
        toggle() {
            this.isActive ? this.close() : this.open();
        }
    }
};
</script>

<style scoped lang="scss">
  @import "./CollapseComponent.scss";
</style>
