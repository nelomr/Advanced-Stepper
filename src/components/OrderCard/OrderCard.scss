.card {
    max-width: 385px;
    margin: 0 auto;
}

.card--text {
    margin-bottom: 20px;
    text-align: left;
}

.card--title,
.card--content-title {
    font-weight: bold;
}

.card--date {
    color: $text-color;
}

.card--content {
    padding: 20px 25px;
    margin-bottom: 20px;
    border: 1px solid $grey-medium;
    border-radius: 6px;
    text-align: left;
}

.card--box {
    padding: 10px 25px;
    margin: 0 -25px;
    border-top: 1px solid $grey-medium;

    &:last-child {
        padding-bottom: 0;
    }
}

.card--content-title {
    margin-bottom: 10px;
}

.card--product {
    display: flex;
    justify-content: space-between;
}

.card--product-title,
.card--product-price {
    flex: 0 1 calc(50% - 10px);
}

.card--product-price {
    text-align: right;
}

.card--button {
    width: 100%;
    text-align: right;

    .button {
        display: inline-block;
        width: 100%;
        padding: 12px 30px;
        font-size: 14px;
        color: $light-color;
        cursor: pointer;
        cursor: hand;
        text-decoration: none;
        text-transform: uppercase;
        background-color: $secondary-color;
        appearance: none;
        border: none;
        border-radius: 4px;

        &.is-small {
            width: auto;
            padding: 6px 10px;
            font-size: 12px;
            background-color: $line-color;
        }
    }

    &.is-transparent {
        margin-top: 20px;
        text-align: center;

        .button {
            width: auto;
            color: #000;
            background-color: transparent;
            border: 1px solid #000;
        }

        span {
            margin-right: 10px;
        }
    }
}
