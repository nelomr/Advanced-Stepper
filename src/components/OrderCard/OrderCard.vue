<template v-if="order">
  <div class="card">
    <div class="card--text">
      <div class="card--title">Datos del pedido nº {{ order.code }}</div>
      <div class="card--date">Fecha pedido: {{ order.orderDate }}</div>
    </div>
    <div class="card--content">
      <div class="card--content-title">Productos</div>
      <div class="card--box">
        <div v-if="order.cuponId" class="card--product">
          <div class="card--product-title">Cupón {{ order.cuponId }}</div>
          <div class="card--product-price">-20 €</div>
        </div>
        <div class="card--product">
          <div class="card--product-title">Recogida/entrega</div>
          <div class="card--product-price">{{ order.totalDeliveryPrice }} €</div>
        </div>
      </div>
      <div class="card--box">
        <div class="card--content-title">Productos Añadidos</div>
        <div
          v-for="(product) in productsDetailOrder"
          :key="product.id" 
          class="card--product"
        >
          <div class="card--product-title">
              <span v-if="product.quantity > 1">{{product.quantity}} x </span> {{ product.name}}
          </div>
          <div class="card--product-price">{{product.price}} €</div>
        </div>
      </div>
      <div class="card--box">
        <div class="card--product">
          <div class="card--product-title">Total</div>
          <div class="card--product-price">{{ order.totalPrice }} €</div>
        </div>
      </div>
    </div>
    <div class="card--button">
      <Collapse-Component :open-default="false">
          <template slot="collapse-component-header">
              <button class="button">Cambiar hora recogida y entrega</button>
          </template>
          <template slot="collapse-component-content">
            <form action="#" v-if="order.pickupDates.length > 0">
              <div
                v-for="(option, index) in order.pickupDates"
                class="input-radio"
                :key="'option'+index"
              >
                <input v-if="index === 0" type="radio" :id="'option'+index" name="date" :value="option.id" checked>
                <input v-else type="radio" :id="'option'+index" name="date" :value="option.id">
                <label :for="'option'+index">{{ option.pickupDate }} - {{ option.pickupStart }} - {{ option.pickupEnd }}</label>
              </div>
              <div class="card--button">
                <button class="button is-small">
                  Cambiar
                </button>
              </div>
            </form>
          </template>
      </Collapse-Component>
    </div>
    <div class="card--button is-transparent">
      <router-link class="button" :to="{name:'orders'}"> <span>&#8592;</span>VOLVER</router-link>
    </div>
  </div>
</template>

<script>
import CollapseComponent from '@/components/CollapseComponent/CollapseComponent';

export default {
  name: 'card',
  components: {
    CollapseComponent
  },
  props: {
    order: {
      type: Object,
      required: true
    }
  },

  created() {
    this.$store.dispatch('getProductsOrder', this.order.productsOrder);
  },

  computed: {
    productsDetailOrder() {
    let details = [];

    if(this.$store.state.productsDetailOrder.length > 0) {
      this.$store.state.productsDetailOrder.forEach(product => {
        this.order.productsOrder.forEach(productOrder => {
          if(product.id === productOrder.productId) {
            details.push({...productOrder, ...product});
          }
        });
      });

      return details;
    }

    //In order to get some data if token api expires
    return [
      {
        'id': '12',
        'quantity': '2',
        'name': 'Chaqueta',
        'price': '12'
      }
    ]
    }
  }
}
</script>

<style scoped lang="scss">
  @import "./OrderCard.scss";
</style>