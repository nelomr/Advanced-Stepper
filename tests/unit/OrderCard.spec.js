import Vue from 'vue';
import Vuex from 'vuex';
import {shallowMount, createLocalVue} from '@vue/test-utils';
import OrderCard from '@/components/OrderCard/OrderCard';
import VueRouter from 'vue-router';
import CollapseComponent from '@/components/CollapseComponent/CollapseComponent.vue';
Vue.component('CollapseComponent', CollapseComponent);

const localVue = createLocalVue();
localVue.use(VueRouter);
localVue.use(Vuex);
const router = new VueRouter();

describe('OrderCard', () => {
    let props;
    let order = {"code":"CO107341","orderDate":"2018-11-09","id":"e29f59ab-2793-4644-bec5-5818fd67a165","logisticCode":"LOCAL","cuponId":"","totalPrice":14500,"totalDeliveryPrice":0,"productsOrder":[{"productId":"b6d1f4e9-85a9-449d-a16e-cb308cbbeb35","quantity":1},{"productId":"8bf0e74b-284f-4a51-bfb1-aa5f1e0ab279","quantity":1},{"productId":"46297509-f50e-4b19-ba6c-bf9744414f37","quantity":2}],"pickupDates":[{"id":"01e6ebbf-99fc-443d-9cb4-a1d0714f2892","pickupDate":"2018-11-13","pickupStart":"17:00:00","pickupEnd":"18:00:00"},{"id":"fe4d70df-7a7c-43dc-b45f-21c7e9ae55e1","pickupDate":"2018-11-09","pickupStart":"17:00:00","pickupEnd":"18:00:00"}]};
    let productsDetailOrder = [{"productId":"46297509-f50e-4b19-ba6c-bf9744414f37","quantity":2,"id":"46297509-f50e-4b19-ba6c-bf9744414f37","name":"PantalÃ³n","price":3500},{"productId":"8bf0e74b-284f-4a51-bfb1-aa5f1e0ab279","quantity":1,"id":"8bf0e74b-284f-4a51-bfb1-aa5f1e0ab279","name":"Saco","price":3500},{"productId":"b6d1f4e9-85a9-449d-a16e-cb308cbbeb35","quantity":1,"id":"b6d1f4e9-85a9-449d-a16e-cb308cbbeb35","name":"chaqueta","price":4000}];
    let store;
    let state;
    let actions;
    
    beforeEach(() => {
        state = {
            productsDetailOrder
        };

        actions = {
            getProductsOrder: jest.fn()
        };

        store = new Vuex.Store({
            state,
            actions
        });

        props = {
            order
        };
    });
    
    const build = () => {
        const wrapper = shallowMount(OrderCard, {
            mocks: {$store: store},
            router,
            computed: {
                productsDetailOrder() {
                    return store.state.productsDetailOrder;
                }
            },
            propsData: props,
            stubs: ['router-link', 'router-view']
        });

        return {
            wrapper
        };
    };

    it('renders the component', () => {
        const {wrapper} = build();

        expect(wrapper.html()).toMatchSnapshot();
    });
});
