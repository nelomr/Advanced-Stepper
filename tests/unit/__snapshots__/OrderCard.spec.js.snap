// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderCard renders the component 1`] = `
<div class="card">
  <div class="card--text">
    <div class="card--title">Datos del pedido nº CO107341</div>
    <div class="card--date">Fecha pedido: 2018-11-09</div>
  </div>
  <div class="card--content">
    <div class="card--content-title">Productos</div>
    <div class="card--box">
      <!---->
      <div class="card--product">
        <div class="card--product-title">Recogida/entrega</div>
        <div class="card--product-price">0 €</div>
      </div>
    </div>
    <div class="card--box">
      <div class="card--content-title">Productos Añadidos</div>
      <div class="card--product">
        <div class="card--product-title"><span>2 x </span> Pantalón
        </div>
        <div class="card--product-price">3500 €</div>
      </div>
      <div class="card--product">
        <div class="card--product-title">
          <!----> Saco
        </div>
        <div class="card--product-price">3500 €</div>
      </div>
      <div class="card--product">
        <div class="card--product-title">
          <!----> chaqueta
        </div>
        <div class="card--product-price">4000 €</div>
      </div>
    </div>
    <div class="card--box">
      <div class="card--product">
        <div class="card--product-title">Total</div>
        <div class="card--product-price">14500 €</div>
      </div>
    </div>
  </div>
  <div class="card--button">
    <collapse-component-stub><template></template> <template></template></collapse-component-stub>
  </div>
  <div class="card--button is-transparent">
    <router-link-stub to="[object Object]" class="button"><span>←</span>VOLVER</router-link-stub>
  </div>
</div>
`;
